test/lib/ansible_test/_internal/host_profiles.py

```diff
@@ -202,31 +202,23 @@ def create_single_host(name: str, variables: dict[str, t.Union[str, int]]) -> In
     def write(self, args: CommonConfig, path: str) -> None:
         """Write the given inventory to the specified path on disk."""
 
-        # NOTE: Switching the inventory generation to write JSON would be nice, but is currently not possible due to the use of hard-coded inventory filenames.
-        #       The name `inventory` works for the POSIX integration tests, but `inventory.winrm` and `inventory.networking` will only parse in INI format.
-        #       If tests are updated to use the `INVENTORY_PATH` environment variable, then this could be changed.
-        #       Also, some tests detect the test type by inspecting the suffix on the inventory filename, which would break if it were changed.
-
-        inventory_text = ''
-
+        inventory_data: dict[str, t.Any] = dict()
```

Comment:
The type annotations can be simplified by being more specific up front:

```suggestion
        inventory_data: dict[str, dict[str, dict[str, dict[str, object]]]] = dict()

```